modules/apps/remote-web-app/remote-web-app-web/src/main/java/com/liferay/remote/web/app/web/internal/RemoteWebAppPortletRegistrar.java

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.remote.web.app.web.internal;
+
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.remote.web.app.model.RemoteWebAppEntry;
+import com.liferay.remote.web.app.service.RemoteWebAppEntryLocalService;
+import com.liferay.remote.web.app.web.internal.portlet.RemoteWebAppPortlet;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.osgi.framework.BundleContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Iván Zaera Avellón
+ */
+@Component(immediate = true, service = RemoteWebAppPortletRegistrar.class)
+public class RemoteWebAppPortletRegistrar {
+
+	public void registerPortlet(RemoteWebAppEntry remoteWebAppEntry) {
+		_registerPortlet(
+			remoteWebAppEntry.getEntryId(), remoteWebAppEntry.getName(),
+			remoteWebAppEntry.getUrl());
+	}
+
+	public void unregisterPortlet(RemoteWebAppEntry remoteWebAppEntry) {
+		_unregisterPortlet(remoteWebAppEntry.getEntryId());
+	}
+
+	@Activate
+	protected void activate(BundleContext bundleContext) {
+		_bundleContext = bundleContext;
+
+		if (_log.isInfoEnabled()) {
+			_log.info("Starting remote web apps");
+		}
+
+		_registerPortlet(0, "El País", "http://www.elpais.com");
+
+		/*
+
+		for (RemoteWebAppEntry remoteWebAppEntry :
```

Comment:
On activation, the service should register every remote web app entry in the DB.